mountpoint-s3/src/fs.rs

```diff
@@ -339,7 +341,13 @@ where
             *request = Some(self.prefetcher.get(&self.bucket, &handle.full_key, handle.object_size));
```

Comment:
I think we should pass the ETag in here, rather than passing it on every `read`. It's the same effect, just a little simpler/faster.